src/tests/roc_core/test_heap_arena.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_core/heap_arena.h"
+#include "roc_core/memory_ops.h"
+
+namespace roc {
+namespace core {
+
+TEST_GROUP(heap_arena) {};
+
+TEST(heap_arena, guard_object) {
+    core::HeapArena::set_flags(core::DefaultHeapArenaFlags
+                               & ~core::HeapArenaFlag_EnableGuards);
```

Comment:
nitpick: we can move setting and restoring flags to setup() and teardown() methods of TEST_GROUP. This way we can be sure that flags are restored no matter if test succeeded or failed.